crates/fuel-gas-price-algorithm/src/v1.rs

```diff
@@ -374,43 +366,60 @@ impl AlgorithmUpdaterV1 {
 
     fn da_block_update(
         &mut self,
-        height: u32,
-        block_bytes: u64,
-        block_cost: u64,
+        height_range: Range<u32>,
+        range_cost: u128,
     ) -> Result<(), Error> {
         let expected = self.da_recorded_block_height.saturating_add(1);
-        if height != expected {
+        let first = height_range.start;
+        if first != expected {
             Err(Error::SkippedDABlock {
-                expected: self.da_recorded_block_height.saturating_add(1),
-                got: height,
+                expected,
+                got: first,
             })
         } else {
-            let new_cost_per_byte: u128 = (block_cost as u128)
-                .checked_div(block_bytes as u128)
-                .ok_or(Error::CouldNotCalculateCostPerByte {
-                    bytes: block_bytes,
-                    cost: block_cost,
-                })?;
-            self.da_recorded_block_height = height;
+            let last = height_range.end.saturating_sub(1);
+            let range_bytes = self.drain_l2_block_bytes_for_range(height_range)?;
+            let new_cost_per_byte: u128 = range_cost.checked_div(range_bytes).ok_or(
+                Error::CouldNotCalculateCostPerByte {
+                    bytes: range_bytes,
+                    cost: range_cost,
+                },
+            )?;
+            self.da_recorded_block_height = last;
             let new_block_cost = self
                 .latest_known_total_da_cost_excess
-                .saturating_add(block_cost as u128);
+                .saturating_add(range_cost);
             self.latest_known_total_da_cost_excess = new_block_cost;
             self.latest_da_cost_per_byte = new_cost_per_byte;
             Ok(())
         }
     }
 
+    fn drain_l2_block_bytes_for_range(
+        &mut self,
+        height_range: Range<u32>,
+    ) -> Result<u128, Error> {
+        let mut total: u128 = 0;
+        for expected_height in height_range {
+            let (actual_height, bytes) = self
+                .unrecorded_blocks
+                .pop_first()
```

Comment:
Could `split_off()` be even more performant here? Not a big deal, because the ranges will most likely be of reasonable size, but worth having a look at.

Reply:
`split_off` wouldn't do checks that all the values we expect would be there. 

It also gives us the wrong values, so we'd have to swap them every time. i.e. it removes the elements after the split point from the map, and we want it to remove the elements before the split point. It would be nice if there was a `split_off_front`. Probably would still be more performant with `split_off` though.

Reply:
`pop_first()` is acceptable.